docs/tutorials/detect.rst

```diff
@@ -0,0 +1,262 @@
+
+When choosing your parameters, you should experiment with different values using your data before running a `detect` run. A good way to do this is to use a combination of the `Archive` and `STALTAOnset` classes to grab sections of data from your archive and apply different filter and STALTA parameters to it. The data can be manipulated and displayed using `matplotlib` as in the example below.
+
+::
+
+    import matplotlib.pyplot as plt
+    from quakemigrate.signal.onset import STALTAOnset
+    from quakemigrate.io import Archive, read_stations
+
+    # define an archive object
+    archive_path = "/path/to/archived/data"
+    station_file = "/path/to/station_file"
+    stations = read_stations(station_file)
+    archive = Archive(archive_path=archive_path, stations=stations,
+                    archive_format="YEAR/JD/STATION")
+                    
+    # read a snippet of data (ideally around a known event)
+    starttime = "2018-001T10:00:00.0"
+    endtime = "2018-001T10:05:00.0"
+    data = archive.read_waveform_data(starttime, endtime, sampling_rate)
+
+    # define the onset function
+    onset = STALTAOnset(position="classic")
+    onset.p_bp_filter = [2, 9.9, 2]
+    onset.s_bp_filter = [2, 9.9, 2]
+    onset.p_onset_win = [0.2, 1.5]
+    onset.s_onset_win = [0.2, 1.5]]
+
+    # apply the onset function to the data snippet
+    onset_data = onset.calculate_onsets(data, log=False)
+
+    # onset data is a numpy array of the P and S onsets
+    # to plot the Z-component of a single station
+    fig, axs = plt.subplots(3, 1)
+
+    raw_waveform = data.signal[0, 0, :]
+    filtered_waveform = data.filtered_signal[0, 0, :]
+    onset_waveform = onset_data[0, :]
+    time = data.times()
+
+    axs[0].plot(time, raw_waveform, 'k-')
+    axs[0].set_xlabel('Time [s]')
+    axs[0].set_title('Raw data')
+
+    axs[1].plot(time, filtered_waveform, 'k-')
+    axs[1].set_xlabel('Time [s]')
+    axs[1].set_title('Filtered data')
+    
+    axs[2].plot(time, raw_waveform, 'k-')
+    axs[2].set_xlabel('Time [s]')
+    axs[2].set_ylabel('signal-to-noise ratio')
+    axs[2].set_title('Onset function')
+
+    plt.tight_layout()
+    plt.show()
+
+`Detect` parameters
+====================
+
+The `detect` stage of QuakeMigrate takes relatively few parameters which the user should set before starting the run. These mostly effect the runtime of the detect run and optimising them can dramtically reduce the overall compute time. 
+
+::
+
+    scan.sampling_rate = 20
+    scan.timestep = 120.
+    scan.threads = 12
+
+The sampling rate should be chosen to be the minimum possible given your chosen filter/signal frequency content as a coalesence grid is calculated for each sample. In this case filtering between 2 - 10 Hz was best so it was possible to decimate 100 sps data to 20 sps. 
```

Comment:
Technically needs to match the sampling rate of the onset data. Changing this a wee bit.